INDI/INDI/indiserver.c

```diff
@@ -156,16 +155,21 @@ typedef struct
     int pid;            /* process id or REMOTEDVR if remote */
     int rfd;            /* read pipe fd */
     int wfd;            /* write pipe fd */
-    int efd;            /* stderr from driver, if local */
     int restarts;       /* times process has been restarted */
     LilXML *lp;         /* XML parsing context */
     FQ *msgq;           /* Msg queue */
     unsigned int nsent; /* bytes of current Msg sent so far */
-} DvrInfo;
+    struct strDvrInfo* pNextToRestart; /* next to restart, or NULL */
+    int restartDelayus; /* Microseconds before next restart attempt */
```

Comment:
This keeps track of the time until the next restart attempt for this DvrInfo structure that has failed communcations in some way.

This value will be set to 10,000,000 (10s) whne the driver communcations fails, and will be decremented after each select(2) call (see indiRun() routine below) by the amount of time it took the select(2) call to run.  When this value reaches 0 or less, INDI server will attempt a restart for this driver.